File: SurveyBuilder/SurveyBuilder.API/Models/Question.cs
namespace SurveyBuilder.API.Models;

public class Question
{
    public Guid Id { get; set; }
    public string Text { get; set; } = string.Empty;
    public QuestionType Type { get; set; }
    public int Order { get; set; }
    public bool IsRequired { get; set; } = false;

    public Guid SurveyId { get; set; }
    public Survey Survey { get; set; } = null!;

    public ICollection<Option> Options { get; set; } = [];
    public ICollection<Answer> Answers { get; set; } = [];
}

File: SurveyBuilder/survey-builder-client/src/types/question.ts
export const QuestionType = {
  Text: 0,
  SingleChoice: 1,
  MultipleChoice: 2,
  Rating: 3,
} as const

export type QuestionType = typeof QuestionType[keyof typeof QuestionType]

export interface Question {
  id: string
  surveyId: string
  text: string
  type: QuestionType
  order: number
  isRequired: boolean
}

export interface Option {
  id: string
  questionId: string
  text: string
  order: number
}

export interface CreateQuestionRequest {
  text: string
  type: QuestionType
  order: number
  isRequired: boolean
}

export interface UpdateQuestionRequest {
  text: string
  type: QuestionType
  order: number
  isRequired: boolean
}

export interface CreateOptionRequest {
  text: string
  order: number
}

export interface UpdateOptionRequest {
  text: string
  order: number
}
